apps/qubit/modules/actor/actions/contextMenuComponent.class.php

```diff
@@ -29,74 +29,47 @@ class ActorContextMenuComponent extends sfComponent
 {
   public function execute($request)
   {
-    if (!isset($request->limit))
-    {
-      $request->limit = sfConfig::get('app_hits_per_page');
-    }
-
-    $this->resource = $request->getAttribute('sf_route')->resource;
+    sfContext::getInstance()->getConfiguration()->loadHelpers(array('Url'));
 
-    // Arrays used to store all the descriptions related to this resource
-    // and the paginated results related by the event type
-    $this->relatedInfoObjects = $this->pagers = array();
+    $page = 1;
+    $limit = sfConfig::get('app_hits_per_page', 10);
 
-    // Get "subject of" information objects (name access point)
-    $criteria = new Criteria;
-    $criteria->add(QubitRelation::OBJECT_ID, $this->resource->id);
-    $criteria->add(QubitRelation::TYPE_ID, QubitTerm::NAME_ACCESS_POINT_ID);
-    $criteria->addJoin(QubitRelation::SUBJECT_ID, QubitInformationObject::ID);
+    $this->lists = array();
 
-    // Filter draft descriptions
-    $criteria = QubitAcl::addFilterDraftsCriteria($criteria);
+    // Subject of
+    $resultSet = ActorRelatedInformationObjectsAction::getRelatedInformationObjects($this->resource->id, $page, $limit);
 
-    // Paginate results
-    $pager = new QubitPager('QubitInformationObject');
-    $pager->setCriteria($criteria);
-    $pager->setMaxPerPage($request->limit);
-    $pager->setPage($request->page);
-
-    $role = __('Subject');
-    if (0 < count($pager->getResults()))
+    if ($resultSet->getTotalHits() > 0)
     {
-      // Add pager only if needed
-      if ($pager->getNbResults() > $request->limit)
-      {
-        $this->pagers[$role] = $pager;
-      }
+      $pager = new QubitSearchPager($resultSet);
+      $pager->setPage($page);
+      $pager->setMaxPerPage($limit);
+      $pager->init();
 
-      $this->relatedInfoObjects[$role] = $pager->getResults();
+      $this->lists[] = array(
```

Comment:
Maybe add a comment here about what $this->lists is for. It isn't immediately obvious what it's for if you're just looking at the one file and not the entire PR for context.